test/e2e/storage/openstack_volume_cluster_ds.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/rand"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere/vclib"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = SIGDescribe("Volume Provisioning On Clustered Datastore [Feature:openstack]", func() {
+	f := framework.NewDefaultFramework("volume-provision")
+
+	var client clientset.Interface
+	var namespace string
+	var scParameters map[string]string
+	var clusterDatastore string
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("openstack")
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+		scParameters = make(map[string]string)
+
+		clusterDatastore = os.Getenv("CLUSTER_DATASTORE")
+		Expect(clusterDatastore).NotTo(BeEmpty(), "Please set CLUSTER_DATASTORE system environment. eg: export CLUSTER_DATASTORE=<cluster_name>/<datastore_name")
+	})
+
+	It("verify static provisioning on clustered datastore", func() {
+		var volumePath string
+		osp, err := getOpenstack(client)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating a test vsphere volume")
+		volumeOptions := new(vclib.VolumeOptions)
+		volumeOptions.CapacityKB = 2097152
+		volumeOptions.Name = "e2e-vmdk-" + namespace
+		volumeOptions.Datastore = clusterDatastore
+
+		tags := map[string]string{
+		"test": "value",
+		}
+		volumeID, _, _, err := osp.CreateVolume("kubernetes-test-volume-"+rand.String(10), 1, "", "", &tags)
```

Comment:
usually a generated name is assigned to a var so that it can be referenced later, eg in an error msg.

Reply:
This was moved to helper function.